Backend/Controllers/FavoureMoviesController.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Backend.Data;
+using Backend.Models;
+using Backend.Data.Abstraction;
+using Backend.Utils;
+using Backend.DTOs;
+
+namespace Backend.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FavouriteMoviesController : ControllerBase
+    {
+        private readonly IFavouriteMovieRepository _repository;
+        private readonly IUserRepository _userRepository;
+
+        public FavouriteMoviesController(IFavouriteMovieRepository repository, IUserRepository userRepository)
+        {
+            _repository = repository;
+            _userRepository = userRepository;
+        }
+
+        // GET: api/FavouriteMovies
+        [HttpGet]
+        public async Task<ActionResult<Dictionary<int, List<Movie>>>> GetUserFavouriteMovies([FromHeader] string emailOrUsername)
```

Comment:
Not sure how FromHeader will work. I am just not sure where to place the headers. Also please add skip, take functionality to this take as well in case a user favorites more than 12 movies we want to have pagination again

Reply:
![image](https://github.com/NoHop3/via-sem6-sep/assets/82092907/1c6a67eb-fd7e-4c95-961a-698834a145f1)